pkg/controller/argocd/deployment.go

```diff
@@ -669,6 +669,10 @@ func (r *ReconcileArgoCD) reconcileRedisDeployment(cr *argoprojv1a1.ArgoCD) erro
 		Env:       proxyEnvVars(),
 	}}
 
+	if err := applyReconcilerHook(cr, deploy); err != nil {
```

Comment:
Should we add a test to verify this behaviour ?

Reply:
added a test to verify the behaviour.

Reply:
The current version `6087379c3dbd275e48110b0be0d59431613c6959` doesn't seem to have this change?

Reply:
yeah it doesn't, it is a part of previous commit. This PR has 2 commits! `22523cb1babcae613531be61180e87bef47ab8b1`